File: wizards/com/sun/star/wizards/web/data/CGSession.java
/*
 * This file is part of the LibreOffice project.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 *
 * This file incorporates work covered by the following license notice:
 *
 *   Licensed to the Apache Software Foundation (ASF) under one or more
 *   contributor license agreements. See the NOTICE file distributed
 *   with this work for additional information regarding copyright
 *   ownership. The ASF licenses this file to you under the Apache
 *   License, Version 2.0 (the "License"); you may not use this file
 *   except in compliance with the License. You may obtain a copy of
 *   the License at http://www.apache.org/licenses/LICENSE-2.0 .
 */
package com.sun.star.wizards.web.data;

import javax.xml.parsers.DocumentBuilderFactory;
import javax.xml.parsers.ParserConfigurationException;

import com.sun.star.wizards.common.*;

import org.w3c.dom.Document;
import org.w3c.dom.Node;

public class CGSession extends ConfigSetItem implements XMLProvider
{

    public String cp_InDirectory;
    public String cp_OutDirectory;
    public String cp_Name;
    public CGContent cp_Content = new CGContent();
    public CGDesign cp_Design = new CGDesign();
    public CGGeneralInfo cp_GeneralInfo = new CGGeneralInfo();
    public ConfigSet cp_Publishing = new ConfigSet(CGPublish.class);
    public CGStyle style; // !!!
    public boolean valid = false;

    public Node createDOM(Node parent)
    {
        Node root = XMLHelper.addElement(parent, "session",
                new String[]
                {
                    "name", "screen-size"
                },
                new String[]
                {
                    cp_Name, getScreenSize()
                });

        //cp_Design.createDOM(root);
        cp_GeneralInfo.createDOM(root);
        //cp_Publishing.createDOM(root);
        cp_Content.createDOM(root);

        return root;
    }

    private String getScreenSize()
    {
        switch (cp_Design.cp_OptimizeDisplaySize)
        {
            case 0:
                return "640";
            case 1:
                return "800";
            case 2:
                return "1024";
            default:
                return "800";
        }
    }

    public CGLayout getLayout()
    {
        return (CGLayout) ((CGSettings) root).cp_Layouts.getElement(cp_Design.cp_Layout);
    }

    public CGStyle getStyle()
    {
        return (CGStyle) ((CGSettings) root).cp_Styles.getElement(cp_Design.cp_Style);
    }

    public void setLayout(short[] layout)
    {
        //dummy
    }

    public Node createDOM()
            throws ParserConfigurationException
    {

        DocumentBuilderFactory factory =
                DocumentBuilderFactory.newInstance();
        Document doc = factory.newDocumentBuilder().newDocument();
        createDOM(doc);
        return doc;
    }
}

File: wizards/com/sun/star/wizards/web/data/CGSession.py
#
# This file is part of the LibreOffice project.
#
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.
#
# This file incorporates work covered by the following license notice:
#
#   Licensed to the Apache Software Foundation (ASF) under one or more
#   contributor license agreements. See the NOTICE file distributed
#   with this work for additional information regarding copyright
#   ownership. The ASF licenses this file to you under the Apache
#   License, Version 2.0 (the "License"); you may not use this file
#   except in compliance with the License. You may obtain a copy of
#   the License at http://www.apache.org/licenses/LICENSE-2.0 .
#
from ...common.ConfigGroup import ConfigGroup
from ..WebConfigSet import WebConfigSet
from ...common.XMLHelper import XMLHelper
from .CGContent import CGContent
from .CGDesign import CGDesign
from .CGGeneralInfo import CGGeneralInfo
from .CGPublish import CGPublish

from com.sun.star.beans import StringPair

class CGSession(ConfigGroup):

    def __init__(self):
        self.cp_Index = -1
        self.cp_InDirectory = str()
        self.cp_OutDirectory = str()
        self.cp_Name = str()
        self.cp_Content = CGContent()
        self.cp_Design = CGDesign()
        self.cp_GeneralInfo = CGGeneralInfo()
        self.cp_Publishing = WebConfigSet(CGPublish)
        self.valid = False

    def createDOM(self, doc):
        root = XMLHelper.addElement(doc, "session",
                                    ["name", "screen-size"],
                                    [self.cp_Name, self.getScreenSize()])
        self.cp_GeneralInfo.createDOM(root)
        self.cp_Content.createDOM(root)
        return root

    def serializeNode(self, node):
        xBuffer = self.root.xmsf.createInstance("com.sun.star.io.Pipe")
        xTextInputStream = self.root.xmsf.createInstance("com.sun.star.io.TextInputStream")
        xSaxWriter = self.root.xmsf.createInstance( "com.sun.star.xml.sax.Writer" )
        xSaxWriter.setOutputStream(xBuffer)
        xTextInputStream.setInputStream(xBuffer)
        node.serialize(xSaxWriter, tuple([StringPair()]))
        result = ""
        while (not xTextInputStream.isEOF()):
            sLine = xTextInputStream.readLine()
            if (not sLine == "") and (not sLine.startswith("<?xml")):
                result = result + sLine + "\n"

    def getScreenSize(self):
        tmp_switch_var1 = self.cp_Design.cp_OptimizeDisplaySize
        if tmp_switch_var1 == 0:
            return "640"
        elif tmp_switch_var1 == 1:
            return "800"
        elif tmp_switch_var1 == 2:
            return "1024"
        else:
            return "800"

    def getLayout(self):
        return self.root.cp_Layouts.getElement(self.cp_Design.cp_Layout)

    def getStyle(self):
        return self.root.cp_Styles.getElement(self.cp_Design.cp_Style)

    def createDOM1(self):
        factory = self.root.xmsf.createInstance("com.sun.star.xml.dom.DocumentBuilder")
        doc = factory.newDocument()
        self.createDOM(doc)
        return doc
